src/CSharp/CodeCracker/Style/StringFormatAnalyzer.cs

```diff
@@ -28,21 +29,24 @@ public class StringFormatAnalyzer : DiagnosticAnalyzer
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
-        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(Analyzer, SyntaxKind.InvocationExpression);
+        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(AnalyzeFormatInvocation, SyntaxKind.InvocationExpression);
 
-        private void Analyzer(SyntaxNodeAnalysisContext context)
+        private static void AnalyzeFormatInvocation(SyntaxNodeAnalysisContext context) =>
+            AnalyzeFormatInvocation(context, "Format", "string.Format(string, ", "string.Format(string, params object[])", Rule);
+
+        public static void AnalyzeFormatInvocation(SyntaxNodeAnalysisContext context, string methodName, string methodOverloadSignature, string methodWithArraySignature, DiagnosticDescriptor rule)
         {
             if (context.IsGenerated()) return;
             var invocationExpression = (InvocationExpressionSyntax)context.Node;
             var memberExpresion = invocationExpression.Expression as MemberAccessExpressionSyntax;
-            if (memberExpresion?.Name?.ToString() != "Format") return;
+            if (memberExpresion?.Name?.ToString() != methodName) return;
             var memberSymbol = context.SemanticModel.GetSymbolInfo(memberExpresion).Symbol;
             if (memberSymbol == null) return;
-            if (!memberSymbol.ToString().StartsWith("string.Format(string, ")) return;
+            if (!memberSymbol.ToString().StartsWith(methodOverloadSignature)) return;
             var argumentList = invocationExpression.ArgumentList as ArgumentListSyntax;
             if (argumentList?.Arguments.Count < 2) return;
             if (!argumentList.Arguments[0]?.Expression?.IsKind(SyntaxKind.StringLiteralExpression) ?? false) return;
-            if (memberSymbol.ToString() == "string.Format(string, params object[])" && argumentList.Arguments.Skip(1).Any(a => context.SemanticModel.GetTypeInfo(a.Expression).Type.TypeKind == TypeKind.Array)) return;
+            if (memberSymbol.ToString() == methodWithArraySignature && argumentList.Arguments.Skip(1).Any(a => context.SemanticModel.GetTypeInfo(a.Expression).Type.TypeKind == TypeKind.Array)) return;
```

Comment:
`argumentList.Arguments.Skip(1).Any(a => context.SemanticModel.GetTypeInfo(a.Expression).Type.TypeKind == TypeKind.Array)` can be computed once